code/modules/mob/living/carbon/human/inventory.dm

```diff
@@ -270,6 +271,7 @@
 					overlay_fullscreen("nearsighted", /obj/screen/fullscreen/impaired, 1)
 			if(G.vision_flags || G.darkness_view || G.invis_override || G.invis_view)
 				update_sight()
+				update_client_colour()

```

Comment:
This should probably be shifted out of the `if` it's currently under, or it won't get called for all glasses.